test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelpersTest.cs

```diff
@@ -148,5 +148,110 @@ public class TagHelpersTests
             var responseContent = await response.Content.ReadAsStringAsync();
             Assert.Equal(expectedContent, responseContent);
         }
+
+        [Fact]
+        public async Task CacheTagHelper_CanCachePortionsOfViewsPartialViewsAndViewComponents()
+        {
+            // Arrange
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+            client.BaseAddress = new Uri("http://localhost");
+            client.DefaultRequestHeaders.Add("Locale", "N");
+
+            // Act - 1
+            // Verify that content gets cached based on vary-by-params
+            var response1 = await client.GetStringAsync("/catalog?categoryId=1&correlationid=1");
+
+            // Assert - 1
+            var expected1 =
+@"<h2>Category: Laptops</h2>
+<h2>Region: North</h2>
+
+    <h2>Cached content</h2>
+    Locations closest to your locale:
+
+NorthWest Store
+<div>CorrelationId in View Component: 1</div>
+
+    <partial-title>Listing items</partial-title>
+
+Cached Content for Laptops
+<div>CorrelationId in Partial: 1</div>
+
+    <div>CorrelationId in Splash: 1</div>";
```

Comment:
we've been using resources under Compiler/Resources for longer expected strings.